### DISCUSSION TITLE
[boost-math] build failure with PCH error

### DISCUSSION THREAD

**Discussion by diehard2** [microsoft/vcpkg#23046] [Q&A]
**Host Environment**
- OS: CentOS8
- Compiler: gcc 8.5

**To Reproduce**
Docker image that reproduces the error. Error "One or more PCH files were found but they were invalid". The suggested fix is turning off pch. I added it to the bjam options file, but it did not appear to be applied. This error does not occur if I use the developer toolkit with gcc-10, but since its the base developers toolkit for CentOS 8 I figure I would raise it here. You can ignore the stuff with Arrow, I just modified another issue to highlight this one.

`FROM centos:latest

RUN yum update -y && yum install -y initscripts;

RUN yum group install -y "Development Tools"

RUN mkdir parquet_test && cd parquet_test && git config --global http.sslverify "false" && git clone https://github.com/microsoft/vcpkg && ./vcpkg/bootstrap-vcpkg.sh

RUN cd parquet_test && SNIPPET="{\"name\": \"versions-test\",\"version\": \"1.0.0\",\"dependencies\": [\"boost-math\"], \"builtin-baseline\": \"39b17cf84c8b389d5800775eaa61e279610c3f33\"}" && echo $SNIPPET > "vcpkg.json"
RUN cd parquet_test && SNIPPET="#include \"arrow/api.h\" \n #include \"parquet/metadata.h\" \n \
    int main(){auto pool = arrow::default_memory_pool(); parquet::FileMetaData* foo; foo->num_columns(); return 0;} \n " && echo -e $SNIPPET > "main.cxx" && mkdir build

RUN cd parquet_test && SNIPPET="cmake_minimum_required(VERSION 3.20) \n \
            set(CMAKE_TOOLCHAIN_FILE \${CMAKE_CURRENT_SOURCE_DIR}/vcpkg/scripts/buildsystems/vcpkg.cmake CACHE STRING \"Vcpkg toolchain file\")  \n \
            project(test) \n \
            find_package(Arrow CONFIG REQUIRED) \n \
            find_package(Parquet CONFIG REQUIRED) \n \
            include_directories(\${ARROW_INCLUDE_DIR}) \n \
            add_executable(test main.cxx) \n \
            target_link_libraries(test PRIVATE \${PARQUET_LIBRARIES} \${ARROW_LIBRARIES}) \n " && echo -e $SNIPPET > "CMakeLists.txt" \
            && cd build && cmake .. `

**Comment by PhoebeHui** [comment]
@diehard2, you are trying to pass C and C++ flag to boost-math to turn off PCH, right? and it doesn't work. Could you try [vcpkg_cxx_flags](https://github.com/microsoft/vcpkg/blob/master/docs/users/triplets.md#vcpkg_cxx_flags) and see if it works for you?

**Comment by diehard2** [comment]
Hi Phoebe, I'm in manifest mode and I attempted to alter the package directly. I'm using this in my cmake file to call vcpkg

set(CMAKE_TOOLCHAIN_FILE ${CMAKE_CURRENT_SOURCE_DIR}/vcpkg/scripts/buildsystems/vcpkg.cmake CACHE STRING "Vcpkg toolchain file")

I have a workaround using the redhat developer kits for gcc10/11, but not everyone is going to have root so I figured I would raise it here

**Comment by PhoebeHui** [comment]
@diehard2, thanks for sharing the case and workaround which may help other user. you can try to add the flag in your triplets, it should work.
